pip/__init__.py

```diff
@@ -181,19 +181,51 @@ def parseopts(args):
     # the subcommand name
     cmd_name = args_else[0]
 
-    if cmd_name not in commands_dict:
-        guess = get_similar_commands(cmd_name)
-
-        msg = ['unknown command "%s"' % cmd_name]
-        if guess:
-            msg.append('maybe you meant "%s"' % guess)
-
-        raise CommandError(' - '.join(msg))
-
     # all the args without the subcommand
     cmd_args = args[:]
     cmd_args.remove(cmd_name)
 
+    # Autocorrect command name
+    if cmd_name not in commands_dict:
+        # MARK: The following should be loaded from the configuration file
```

Comment:
This part probably depends on #4192. I'm willing to shoot up another PR (or update this one) once that merges.